src/executors/linux_android.yml

```
@@ -12,7 +12,7 @@ parameters:
     type: string
     default: '5.1'
   resource_class:
-    description: Changes the resource class of the executor. Requires a support request to enable the resource_class parameter. See https://circleci.com/docs/2.0/configuration-reference/#resource_class
+    description: You can use whatever resource class your plan has access to. https://circleci.com/product/features/resource-classes/
```

I think [linking to the documentation](https://circleci.com/docs/configuration-reference/#resourceclass) is better because you can find the correct `resource_class` string immediately, compared to the product (marketing) page.

```suggestion
    description: You can use whatever resource class your plan has access to. See https://circleci.com/docs/configuration-reference/#resourceclass
```

Other than that, this is a good contribution – thank you!

---

Can you split this into 2 commits and update the commit messages using [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)?

For example:

```
docs: update description for resource class parameter
```

And:

```
feat: change default macos executor to macos.x86.medium.gen2

BREAKING CHANGE: the `medium` executor is deprecated
```

---

Great idea on the docs link instead of the product page!

I've made the necessary changes as far as the commits go and linked to the MacOS docs page for the iOS job and macos executor.